src/main/java/racingcar/pattern/NumberGenerator.java

```diff
@@ -0,0 +1,8 @@
+package racingcar.pattern;
+
+public interface NumberGenerator {
```

Comment:
전략패턴을 활용해주셨네요 👍👍 

```suggestion
public interface Moveable {
        boolean canMove();
}
```
지금도 잘해주셨지만
위와 같은 인터페이스를 활용한다면 확장에 더 좋다고 생각해요.
단순히 숫자기반으로 움직이는지 판단할 수 있을 뿐만 아니라
시간기반 등 처럼 다양한 경우에도 대처할 수 있을 것 같아요 :slightly_smiling_face: